src/main/java/xyz/kmahyyg/eshopdemo/controller/GoodRestController.java

```diff
@@ -0,0 +1,22 @@
+package xyz.kmahyyg.eshopdemo.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import xyz.kmahyyg.eshopdemo.common.PublicResponse;
+import xyz.kmahyyg.eshopdemo.enums.ErrorStatusEnum;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@RestController
+public class GoodRestController {
+    @PostMapping("/api/search")
+    void search(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String keyword = request.getParameter("q");
+        response.sendRedirect("/show/good/search?itemName="+keyword);
```

Comment:
Please use `try` and `catch` to process the unhandled exception or just remove the `throws` statement.

Reply:
Also this parameter `q` is shown in template, which might introduce reflective XSS here. Escape this input when you show it in the template.